src/main/java/se/sundsvall/chat/api/MessageResource.java

```diff
@@ -0,0 +1,122 @@
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestPart;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+import org.zalando.problem.Problem;
+import org.zalando.problem.violations.ConstraintViolationProblem;
+import se.sundsvall.chat.api.model.Message;
+import se.sundsvall.chat.api.model.MessageRequest;
+import se.sundsvall.chat.service.MessageService;
+import se.sundsvall.dept44.common.validators.annotation.ValidMunicipalityId;
+import se.sundsvall.dept44.common.validators.annotation.ValidUuid;
+
+@RestController
+@Validated
+@RequestMapping("/{municipalityId}/{namespace}/conversations/{conversationId}/messages")
+@ApiResponse(responseCode = "400", description = "Bad request", content = @Content(mediaType = APPLICATION_PROBLEM_JSON_VALUE, schema = @Schema(oneOf = {
+	Problem.class, ConstraintViolationProblem.class
+})))
+@ApiResponse(responseCode = "404", description = "Not Found", content = @Content(mediaType = APPLICATION_PROBLEM_JSON_VALUE, schema = @Schema(implementation = Problem.class)))
+@ApiResponse(responseCode = "500", description = "Internal Server error", content = @Content(mediaType = APPLICATION_PROBLEM_JSON_VALUE, schema = @Schema(implementation = Problem.class)))
+@Tag(name = "Message resource", description = "Message operations")
+class MessageResource {
+
+	private final MessageService service;
+
+	MessageResource(final MessageService service) {
+		this.service = service;
+	}
+
+	@PostMapping(consumes = MULTIPART_FORM_DATA_VALUE, produces = ALL_VALUE)
+	@Operation(description = "Post a message with attachment", responses = {
+		@ApiResponse(responseCode = "201", description = "Created - Successful operation", useReturnTypeSchema = true)
+	})
+	ResponseEntity<Void> postMessage(
+		@Parameter(name = "namespace", description = "Namespace", example = "MY_NAMESPACE") @Pattern(regexp = NAMESPACE_REGEXP, message = NAMESPACE_VALIDATION_MESSAGE) @PathVariable final String namespace,
+		@Parameter(name = "municipalityId", description = "Municipality ID", example = "2281") @ValidMunicipalityId @PathVariable final String municipalityId,
+		@Parameter(name = "conversationId", description = "Conversation ID", example = "b82bd8ac-1507-4d9a-958d-369261eecc15") @ValidUuid @PathVariable final String conversationId,
+		@Parameter(name = "message", description = "Message to be posted") @RequestPart("message") final MessageRequest message,
```

Comment:
`@RequestPart`?

Reply:
I cant find any swagger annotation for `@RequestPart`?

Reply:
I think a meant `@RequestBody` for this comment.